test/unit/rules/no-action-on-submit-button-test.js

```diff
@@ -0,0 +1,203 @@
+import generateRuleTests from '../../helpers/rule-test-harness.js';
+
+generateRuleTests({
+  name: 'no-action-on-submit-button',
+
+  config: true,
+
+  good: [
+    // valid buttons with "button" type, in a form
+    '<form><button type="button" /></form>',
+    '<form><button type="button" {{action this.handleClick}} /></form>',
+    '<form><button type="button" {{action this.handleClick on="click"}} /></form>',
+    '<form><button type="button" {{action this.handleMouseover on="mouseOver"}} /></form>',
+    '<form><button type="button" {{on "click" this.handleClick}} /></form>',
+    '<form><button type="button" {{on "mouseover" this.handleMouseover}} /></form>',
+
+    // valid buttons with "submit" type, in a form
+    '<form><button /></form>',
+    '<form><button type="submit" /></form>',
+    '<form><button type="submit" {{action this.handleMouseover on="mouseOver"}} /></form>',
+    '<form><button type="submit" {{on "mouseover" this.handleMouseover}} /></form>',
+
+    // valid div elements, in a form
+    '<form><div/></form>',
+    '<form><div></div></form>',
+    '<form><div type="submit"></div></form>',
+    '<form><div type="submit" {{action this.handleClick}}></div></form>',
+    '<form><div type="submit" {{on "click" this.handleClick}}></div></form>',
+
+    // valid buttons, only outside a form
+    '<button {{action this.handleClick}} />',
+    '<button {{action this.handleClick on="click"}}/>',
+    '<button {{on "click" this.handleClick}} />',
+    '<button type="submit" {{action this.handleClick}} />',
+    '<button type="submit" {{action this.handleClick on="click"}} />',
+    '<button type="submit" {{action (fn this.someAction "foo")}} />',
+    '<button type="submit" {{on "click" this.handleClick}} />',
+    '<button type="submit" {{on "click" (fn this.addNumber 123)}} />',
+  ],
+
+  bad: [
```

Comment:
Shouldn't we add a test for a deeply-nested button like this to make sure we aren't only checking the direct parent?

* `<form><div><button {{action this.handleClick}} /></div></form>`

Reply:
Good idea! I added it as last scenario ✅